.github/workflows/nightly-build.yml

```diff
@@ -0,0 +1,68 @@
+name: Nightly Build
+
+on:
+  workflow_dispatch: # to be able to run it manually
+  schedule:
+    - cron: '0 0 * * *' # once a day
+
+jobs:
+  evaluateChanges:
+    name: Evaluate changes for run or skipping nightly build
+    runs-on: ubuntu-latest
+    outputs:
+      SHOULD_BUILD: ${{ steps.check.outputs.shouldBuild }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Check
+        id: check
+        run: |
+          if [[ $(git rev-list --after="24 hours" main) ]]; then
+            echo ::set-output name=shouldBuild::true
+          else
+            echo ::set-output name=shouldBuild::false
+          fi
+
+
+  build:
+    name: Generate Nightly Build
+    runs-on: ubuntu-latest
+    needs: evaluateChanges
+    if: ${{ needs.evaluateChanges.outputs.SHOULD_BUILD == 'true' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Build and send apks to telegram
+        run: |
+          for TASK in assembleWebsiteNightly; do
```

Comment:
Minor: move `assembleWebsiteNightly` to an env var as `BUILD_TASKS` or equivalent?